.github/workflows/lint.yml

```diff
@@ -20,11 +12,11 @@ jobs:
         uses: actions/setup-go@v5.0.0
         with:
           go-version: 1.20.x
-
       - name: Checkout code
         uses: actions/checkout@v4.1.1
-
+        with:
+          fetch-depth: 0
       - name: Lint
-        uses: golangci/golangci-lint-action@v3
+        uses: bane/golangci-lint-action@v4.0.0
```

Comment:
we should host this under ethernal's account

Reply:
I don't understand why we need to fork the repository containing the linter GH action. Couldn't we use it from here: https://github.com/golangci/golangci-lint-action?

Reply:
We don't need and will never resort to such measures as long as GitHub actions function correctly as intended. However, in this scenario, they haven't released a new version of the action that utilizes `node20`. In the meantime, they have released an official `v4.0.0` version, which does utilize `node20`. Hence, now we can revert to using them.